src/compass/utils/ale.py

```diff
@@ -0,0 +1,325 @@
+
+    # Oversample CSLC subset and get amplitude
+    img_ovs = isce3.cal.point_target_info.oversample(
+        img, ovs_factor)
+    idx_peak_ovs = np.argmax(np.abs(img_ovs))
+    img_peak_ovs = np.unravel_index(idx_peak_ovs, img_ovs.shape)
+
+    # Get location of the peak wrt the upper left corner of image subset
+    x_peak = x_loc - margin + img_peak_ovs[1] / ovs_factor
+    y_peak = y_loc - margin + img_peak_ovs[0] / ovs_factor
+
+    return x_peak, y_peak
+
+
+def get_cslc(cslc_file, pol='VV'):
+    '''
+    Get CSLC-S1 array associated to 'pol'
+
+    Parameters
+    ----------
+    cslc_file: str
+        File path to CSLC-S1 product
+    pol: str
+        Polarization string of CSLC-S1 array
+        to extract from product
+
+    Returns
+    -------
+    cslc: np.ndarray
+        CSLC-S1 image associated to polarization
+        channel 'pol'
+    '''
+    with h5py.File(cslc_file, 'r') as h5:
+        cslc = h5[f'science/SENTINEL1/CSLC/grids/{pol}'][()]
+    return cslc
+
+
+def get_xy_info(cslc_file):
+    '''
+    Get X/Y-coordinate vector and spacings
+    from CSLC-S1 product in 'cslc_file'
+
+    Parameters
+    ----------
+    cslc_file: str
+        File path to CSLC-S1 product
+
+    Returns
+    -------
+    x_vect: np.ndarray
+        Array of X-coordinates associated to CSLC-S1
+    y_vect: np.ndarray
+        Array of Y-coordinates associated to CSLC-S1
+    x_spac: np.float
+        CSLC-S1 spacing along X-direction
+    y_spac: np.float
+        CSLC-S1 spacing along Y-direction
+    '''
+
+    with h5py.File(cslc_file, 'r') as h5:
+        x_vect = h5['science/SENTINEL1/CSLC/grids/x_coordinates'][()]
+        y_vect = h5['science/SENTINEL1/CSLC/grids/y_coordinates'][()]
+        x_spac = h5['science/SENTINEL1/CSLC/grids/x_spacing'][()]
+        y_spac = h5['science/SENTINEL1/CSLC/grids/y_spacing'][()]
+
+    return x_vect, x_spac, y_vect, y_spac
+
+
+def latlon2utm(lat, lon, out_epsg):
+    '''
+    Converts lat/lon to x/y coordinates
+    specified by 'out_epsg'
+
+    Parameters
+    ----------
+    lat: np.float
+        Latitude coordinates
+    lon: np.float
+        Longitude coordinates
+    out_epsg: int
+        EPSG code identifying coordinate system
+        for lat/lon conversion
+
+    Returns
+    -------
+    x: np.float
+        X-coordinate
+    y: np.float
+        Y-coordinate
+    '''
+
+    _proj = Proj(CRS.from_epsg(out_epsg))
+    x, y = _proj(lon, lat, inverse=False)
+    return x, y
+
+
+def get_clsc_polygon(cslc_file):
+    '''
+    Get the polygon containing the valid values of
+    the CSLC-S1 product in 'cslc_file'
+
+    Parameters
+    ----------
+    cslc_file: str
+        File path to the CSLC-S1 product
+
+    Returns
+    -------
+    cslc_poly: shapely.Polygon
+        Shapely polygon including CSLC-S1 valid values
+    '''
+    with h5py.File(cslc_file, 'r') as h5:
+        poly = h5['science/SENTINEL1/identification/bounding_polygon'][()]
+    cslc_poly = wkt.loads(poly)
```

Comment:
you probably need to decode

```suggestion
    cslc_poly = wkt.loads(poly.decode('utf-8'))
```